.claude/scripts/worktree-claim.test.sh

```
@@ -822,5 +822,390 @@ check "failed default discovery still claims successfully" 0 "$symref_rc" \
+# Naming both SHAs is what makes the warning actionable rather than vague: the operator can see
+# which head they are on and which one they wanted.
+check "the stale-cache warning names the cached AND the origin sha" 0 "$stalecache_rc" \
+  "$stalecache_out" "${stalecache_cached:0:10}"
+
+# The complement, and the reason this cannot just always warn: when the fetch SUCCEEDS the cached
+# ref is current, so the warning must stay silent or it fires on every ordinary attach.
+freshcache_consumer="$tmp/freshcache-consumer"
+git clone -q "$stalecache_origin" "$freshcache_consumer"
+freshcache_rc=0
+freshcache_out="$("$script" add \
+  "$freshcache_consumer" "$tmp/wt-freshcache" "claim-stalecache" "session-freshcache" 2>&1)" || freshcache_rc=$?
```

Shellcheck (0.11.0)

[warning] freshcache_rc appears unused. Verify use (or export if used externally).

(SC2034)